crates/holochain_types/src/dna/dna_bundle.rs

```diff
@@ -237,27 +227,34 @@ impl DnaBundle {
 
 pub(super) async fn hash_bytes(
     zomes: impl Iterator<Item = ZomeManifest>,
-    resources: &mut HashMap<Location, ResourceBytes>,
-) -> DnaResult<Vec<(ZomeName, WasmHash, DnaWasm, Vec<ZomeName>, Option<PathBuf>)>> {
+    resources: &mut HashMap<&ResourceIdentifier, &ResourceBytes>,
+) -> DnaResult<Vec<(ZomeName, WasmHash, DnaWasm, Vec<ZomeName>)>> {
     let iter = zomes.map(|z| {
-        let bytes = resources
-            .remove(&z.location)
-            .expect("resource referenced in manifest must exist");
+        // TODO A bad bundle can cause a crash here, unaceptable!
```

Comment:
So should we fix this to error instead of panicing?

Reply:
Yes, I meant to open an issue to follow up on this https://github.com/holochain/holochain/issues/4927

It's not so serious if you're installing apps from well-intentioned developers because the tooling won't let you create an invalid bundle but it's still worth fixing.